overlord/snapstate/backend.go

```diff
@@ -47,7 +47,7 @@ type StoreService interface {
 	WriteCatalogs(ctx context.Context, names io.Writer, adder store.SnapAdder) error
 
 	Download(context.Context, string, string, *snap.DownloadInfo, progress.Meter, *auth.UserState, *store.DownloadOptions) error
-	DownloadStream(context.Context, string, *snap.DownloadInfo, *auth.UserState) (io.ReadCloser, error)
+	DownloadStream(context.Context, string, *snap.DownloadInfo, int64, *auth.UserState) (io.ReadCloser, int, error)
```

Comment:
The int return value should get a name. I had to follow the implementation to deduce this is a status code. Alternatively get a type for http status code.

Reply:
I will do this in a followup to avoid the extra spread run hassle.